usefulUtilities/botinator/assignmentClientManager.js

```diff
@@ -0,0 +1,400 @@
+                });
+                Messages.sendMessage(ASSIGNMENT_CLIENT_MESSANGER_CHANNEL, messageToSend);
+                break;
+
+
+            case "ARE_YOU_THERE_MANAGER_ITS_ME_BOT":
+                Messages.sendMessage(ASSIGNMENT_MANAGER_CHANNEL, JSON.stringify({
+                    action: "REGISTER_MANAGER",
+                    uuid: sender
+                }));
+                break;
+
+                
+            default:
+                console.log("unrecongized action in assignmentClientManger.js");
+                break;
+        }
+    }
+
+
+    function onTabletChannelMessageReceived(channel, message, sender){
+        try {
+            message = JSON.parse(message);
+        } catch (error) {
+            console.log("invalid object");
+            console.log(error);
+            return;
+        }
+
+        if (channel !== ASSIGNMENT_CLIENT_MESSANGER_CHANNEL || sender === Agent.sessionUUID) {
+            return;
+        }
+
+        switch (message.action) {
+            case "SEND_DATA":
+                if (isPlaying) {
+                    stopAllBots();
+                    isPlaying = false;
+                }   
+
+                if (JSON.stringify(contentBoundaryCorners) !== 
+                    JSON.stringify(message.contentBoundaryCorners)) {
+                    contentBoundaryCorners = message.contentBoundaryCorners;
+                    updateAllBotsPosition();
+                }
+
+                if (volume !== message.volume) {
+                    volume = message.volume;
+                    updateAllBotsVolume();
+                }
+
+                totalNumberOfBotsNeeded = message.totalNumberOfBotsNeeded;
+                break;
+            case "PLAY":
+                if (isPlaying) {
+                    return;
+                }
+                isPlaying = true;
+                startSequence();
+                break;
+            case "STOP":
+                if (isPlaying) {
+                    stopAllBots();
+                }
+                break;
+            case "GET_MANAGER_STATUS":
+                var messageToSend = JSON.stringify({
+                    action: "GET_MANAGER_STATUS",
+                    newAvailableACs: availableAssignmentClientPlayers.length,
+                    isPlaying: isPlaying
+                });
+                Messages.sendMessage(ASSIGNMENT_CLIENT_MESSANGER_CHANNEL, messageToSend);
+                break;
+            default:
+                console.log("unrecongized action in assignmentClientManger.js");
+                break;
+        }
+    }
+
+
+    // #endregion
+    // *************************************
+    // END MESSAGES
+    // *************************************
+
+    // *************************************
+    // START MAIN
+    // *************************************
+    // #region MAIN
+    
+    
+    // Startup for the manager when it comes online
+    function startUp(){
```

Comment:
```suggestion
    function startUp() {
```

please look for other instances of this fix in your files and modify them :)

Reply:
donzo